website/content/docs/api-gateway/configuration/gatewayclassconfig.mdx

```diff
@@ -0,0 +1,196 @@
+---
+layout: docs
+page_title: Consul API Gateway GatewayClassConfig
+description:  >-
+  Consul API Gateway GatewayClassConfig
+---
+
+# GatewayClassConfig
+
+This topic provides full details about the `GatewayClassConfig` resource
+
+## Introduction
+
+The `GatewayClassConfig` object contains Consul API Gateway-related configuration parameters. Apply the parameters by adding the [`GatewayClass`](#gatewayclass) object to your Kubernetes values file and specifying the name of the `GatewayClassConfig`.
+
+## Configuration model
+
+* [`consul`](#consul): object | optional
+  * [`address`](#consul-address): string | optional
+  * [`authentication`](#consul-authentication): object | optional
+    * [`account`]([#consul-authentication-account): string | optional
+    * [`managed`](#consul-authentication-managed): bool  | optional
+    * [`method`](#consul-authentication-method): string | optional
+    * [`namespace`](#consul-authentication-namespace): string  | optional
+  * [`ports`](#consul-ports): object  | optional
+    * [`grpc`](#consul-ports-grpc): integer | optional
+    * [`http`](#consul-ports-http): integer | optional
+  * [`scheme`](#consul-scheme): string  | optional
+* [`copyAnnotations`](#copyAnnotations): object | optional
+  * [`service`](#copyAnnotations-service): array of strings | optional
+* [`deployment`](#deployment): object | optional
+  * [`defaultInstances`](#deployment-defaultinstances): integer | optional
+  * [`maxInstances`](#deployment-maxinstances): integer | optional
+  * [`minInstances`](#deployment-mininstances): integer | optional
+* [`image`](#image): object | optional
+  * [`consulAPIGateway`](#image-consulapigateway): string | optional
+  * [`envoy`](#image-envoy): string | optional
+* [`logLevel`](#loglevel): string | optional
+* [`nodeSelector`](#nodeselector): string | optional
+* [`serviceType`](#servicetype): string | optional
+* [`useHostPorts`](#usehostports): boolean | optional
+
+## Specification
+
+This topic provides details about the configuration parameters.
+
+### consul
+* Type: object
+* Required: optional
+
+### consul.address
+Specifies the address of the Consul server that the `Gateway` communicates with in the gateway pod. If unspecified, the pod attempts to use a local agent on the host where the pod is running.
+* Type: string
+* Required: optional
+* Default: local agent
+
+### consul.authentication.account
+Specifies the Kubernetes service account to use for authentication.
+* Type: string
+* Required: optional
+
+### consul.authentication.managed
+Set to `true` to enable deployments to run with managed service accounts created by the gateway controller. The `consul.authentication.account` field is ignored when this option is enabled.
```

Comment:
Hi, I'm not actually familiar with how this piece works and just copy pasted what we already had. Maybe @andrewstucki, @mikemorris or @nathancoleman will be able to chime in here, but I will try and find the answer for you in the meantime.

Reply:
I don't think this would be at all obvious to a user, and I'm also unsure if it's ever intended to be configured directly - I can't find any reference to _using_ this in the codebase, it appears to be related to [Helm chart configuration](https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/templates/api-gateway-gatewayclassconfig.yaml#L14-L18) and may just be some way to persist that config?